src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,126 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public int getHash() {
+            return hash;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+        public Node<K, V> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<K, V> next) {
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && Objects.equals(node.key, key)) {
+                node.value = value; // Заміняємо значення, якщо ключ уже є
+                return;
+            }
+        }
 
+        addNode(hash, key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && Objects.equals(node.key, key)) {
+                return node.value;
+            }
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addNode(int hash, K key, V value, int index) {
```

Comment:
The `addNode` method is a utility method that is only used inside the `MyHashMap` class. Therefore, it should not be public. You should change its access modifier to `private` to adhere to the encapsulation principle.